src/widgets/span/lv_span.h

```diff
@@ -239,6 +250,40 @@ uint32_t lv_spangroup_get_expand_width(lv_obj_t * obj, uint32_t max_width);
  */
 int32_t lv_spangroup_get_expand_height(lv_obj_t * obj, int32_t width);
 
+/**
+ * Get the span's coords in the spangroup.
+ * @note Before calling this function, please make sure that the layout of span group has been updated.
+ * +--------+
+ * |Heading +--->------------------+
+ * |  Pos   |   |     Heading      |
+ * +--------+---+------------------+
+ * |                               |
+ * |                               |
+ * |                               |
+ * |            Middle   +--------+|
+ * |                     |Trailing||
+ * |                   +-|  Pos   ||
+ * |                   | +--------+|
+ * +-------------------v-----------+
+ * |     Trailing      |
+ * +-------------------+
+ * @param obj       pointer to a spangroup object.
+ * @param span      pointer to a span.
+ * @return the span's coords in the spangroup.
+ */
+lv_span_coords_t lv_spangroup_get_span_coords(lv_obj_t * obj, lv_span_t * span);
+
+/**
+ * Add an event callback to a span.
+ * @param obj       pointer to a spangroup object.
+ * @param span      pointer to a span.
+ * @param cb        the callback function.
+ * @param code      the event code.
+ * @param user_data user data for the callback.
+ */
+void lv_spangroup_add_span_event_cb(lv_obj_t * obj, lv_span_t * span, lv_event_cb_t cb, lv_event_code_t code,
```

Comment:
I would handle it like this:
```
lv_obj_add_event(spangroup, my_event_cb, LV_EVENT_PRESSED/CLICKED/etc, NULL);

void my_event_cb(lv_event_t * e) 
{
   lv_point_t p;
   lv_indev_get_point(lv_indev_get_active(), &p);

  lv_span_t * span = lv_spangroup_get_span_by_point(spangroup, &p);
}
```

I find it more flexible.

Reply:
@kisvegabor I think the method you provide seems appropriate, but when there is a lot of spans, finding efficiency may be a problem. If the trigger event is detected according to the span of the registered time, it will be much more efficient.

Reply:
However you are using `lv_spangroup_get_span_coords` in `lv_spangroup_event` anyways. 
All I meant is letting the user call `get_by_point` function from any events (has basically the same traversing logic that you already has). If the user can do the call in any events it's more flexible. 

Do the performance would be the same.

Reply:
Well, it does look more general, and I will add the relevant API.

Reply:
> I would handle it like this:
> 
> ```c
> lv_obj_add_event(spangroup, my_event_cb, LV_EVENT_PRESSED/CLICKED/etc, NULL);
> 
> void my_event_cb(lv_event_t * e) 
> {
>    lv_point_t p;
>    lv_indev_get_point(lv_indev_get_active(), &p);
> 
>   lv_span_t * span = lv_spangroup_get_span_by_point(spangroup, &p);
> }
> ```
> 
> I find it more flexible.

so, Is this point relative to the coordinates of the component or the global coordinates?

Reply:
I have added a `lv_spangroup_get_span_by_point` now, it returns a span with a relative point of the spangroup obj itself.

Reply:
Great! I mean exactly this!

By having this, I think we don't need `lv_spangroup_add_span_event_cb`.

Reply:
Done